tests/cpp/test_tutorial.cpp

```diff
@@ -1250,4 +1250,204 @@ TEST_F(Tutorial, BasicTMA) {
+    fusion->printMath();
+    fusion->printKernel();
+  }
+
+  constexpr int dim0 = 32, dim1 = 2, dim2 = 4, dim3 = 256;
+  auto options = at::TensorOptions().dtype(dtype).device(at::kCUDA, 0);
+  at::Tensor at_tv0 = at::randn({dim1, dim2, dim3}, options);
+  at::Tensor at_tv1 = at::randn({dim0, dim1, dim2, dim3}, options);
+
+  // Compile with FusionExecutor directly to avoid scheduling
+  FusionExecutor fe;
+  fe.compileFusion(fusion.get(), {at_tv0, at_tv1}, {}, index32bit);
+  auto outputs = fe.runFusion({at_tv0, at_tv1});
+
+  auto at_output = at_tv0 + at_tv1;
+  testValidate(
+      fusion.get(), outputs, {at_tv0, at_tv1}, {at_output}, __LINE__, __FILE__);
+}
+
+TEST_F(Tutorial, VectorizeStorePointwiseTMA) {
+  CompileParams index32bit{DataType::Int32, 255, false};
+
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  constexpr at::ScalarType dtype = at::ScalarType::Float;
+
+  auto tv0 = makeContigTensor(2, aten_to_data_type(dtype));
+  auto tv1 = makeContigTensor(2, aten_to_data_type(dtype));
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = add(tv0, tv1);
+  fusion->addOutput(tv2);
+
+  // Create cache_tvs
+  auto tv0a = tv0->cacheAfter(LoadStoreOpType::CpAsyncBulkTensorTile);
+  auto tv1a = tv1->cacheAfter(LoadStoreOpType::CpAsyncBulkTensorTile);
+  tv2->cacheBefore();
+
+  tv0a->setMemoryType(MemoryType::Shared);
+  tv1a->setMemoryType(MemoryType::Shared);
+
+  auto reference_tv = tv2;
+
+  // Step 1: Create tma domain
+  // Use the root domain as TMA domain
+  //   root domain: [I0, I1]
+
+  constexpr int64_t num_threads = 128;
+  constexpr int64_t vectorization = 2;
+  constexpr int64_t tma_tile = num_threads * vectorization;
+  constexpr int64_t num_stages = 4;
+  constexpr int64_t num_ctas_for_hopper = 132;
+
+  // Step 2: Create Box
+  // After TMA domain creation
+  //         split: [I0, I3, 256]
+  reference_tv->split(-1, tma_tile);
+  //         split: [I2, 4, I3, 256]
+  reference_tv->split(0, num_stages);
+
+  // Step 3: Create Tile
+  // Do nothing here because box == tile
+
+  // Step 4: Schedule Shared Memory Tensor
+  //         split: [I2, 4, I3, 128, 2]
+  reference_tv->split(-1, vectorization);
+  //         split: [I4, 132, 4, I3, 128, 2]
+  reference_tv->split(0, num_ctas_for_hopper);
+  //         reorder: [I4, 132, I3, 4, 128, 2]
+  reference_tv->reorder({{3, 2}, {2, 3}});
+
+  // Transform Operations between cache operations and output reference
+  TransformPropagator propagator(reference_tv);
+  MaxRootDomainInfoSpanningTree(reference_tv).traverse(&propagator);
+
+  // Propagate common parallel dimensions
+  reference_tv->axis(1)->parallelize(ParallelType::BIDx);
+  scheduler_utils::parallelizeAllLike(reference_tv);
+
+  // Vectorization for writing results to gmem
+  reference_tv->axis(-3)->parallelize(ParallelType::Unroll);
+  reference_tv->axis(-2)->parallelize(ParallelType::TIDx);
+  reference_tv->axis(-1)->parallelize(ParallelType::Vectorize);
+
```

Comment:
Should we also do `tv2b->axis(-2)->parallelize(ParallelType::TIDx);`? The generated code does not look right to me:

```C++
...
      for(nvfuser_index_t i30 = 0; i30 < 4; ++i30) {
        nvfuser_index_t i31;
        i31 = 256LL * i30;
        #pragma unroll
        for(nvfuser_index_t i32 = 0; i32 < 128; ++i32) {
          nvfuser_index_t i33;
          i33 = i31 + (2LL * i32);
          #pragma unroll
          for(nvfuser_index_t i34 = 0; i34 < 2; ++i34) {
            nvfuser_index_t i35;
            i35 = i33 + i34;
            T5[i35]
              = T3[i35]
              + T4[i35];
          }
        }
      }
...
```

I think you should either inline to position -2, or manually parallelize it as TIDx